### PR TITLE
Fixes #38

### LINKED ISSUE #38: MapAssert for Vavr's io.vavr.collection.Map
Create `MapAssert` modelled as existing `org.assertj.core.api.MapAssert` with more accurate API than `IterableAssert` that is currently applied by default.

### PR DESCRIPTION
**Pull request by mchmielarz**

### CODE REVIEW

File: src/main/java/org/assertj/vavr/internal/Maps.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+     */
+    public <K, V> void assertDoesNotContain(AssertionInfo info, Map<K, V> actual,
+                                            Tuple2<K, V>[] entries) {
+        failIfNullOrEmpty(entries);
+        assertNotNull(info, actual);
+        failIfEmptySinceActualIsNotEmpty(entries);
+        final Set<Tuple2<? extends K, ? extends V>> found = Array
+                .of(entries)
+                .foldLeft(HashSet.empty(), (set, tuple) -> {
+                    if (actual.contains(tuple)) {
```
Comment (pivovarit) [inline src/main/java/org/assertj/vavr/internal/Maps.java]: If actual contains tuple then add it to set and return? 🤔

Comment (mchmielarz) [reply src/main/java/org/assertj/vavr/internal/Maps.java]: That's the folding function. For a single run of it, we check whether `actual` contains a given `tuple`. If so, then we're collecting it and returning accumulator from the function.

Comment (pivovarit) [reply src/main/java/org/assertj/vavr/internal/Maps.java]: Mixed two different collections 🤕

Comment (mchmielarz) [reply src/main/java/org/assertj/vavr/internal/Maps.java]: Yup, easy to do that with lambdas

**Pull request merged**